exercises/exercise3.js

```diff
@@ -1,12 +1,419 @@
+    handler: function() {
+      return 'ASL....'
+      + '\nJust kidding! That\'d be gross'
+      + '\nTell me where you are by typing \'Okay, my location is...\'and I\'ll tell you about the Native plants in the area...'
+      + '\n(right now I\'m still bound to location only... but in the future I\'ll be able to identify plants based on things like petal shape, location, colour and seeds)';
+      },
+    isListening: true,
+    isCaseSensitive: false,
+  },
+  {
+    label: 'Bot explained script 2',
+    prompt: 'How?',
+    handler: function() {
+      return 'Tell me where you are by typing \'My location is...\' and I\'ll tell you about the Native plants in the area...'
+      + '\n(right now I\'m still bound to location only... but in the future I\'ll be able to identify plants based on things like petal shape, location, colour and seeds)';
+    },
+    // isReply: true,
+    isListening: true,
+    isCaseSensitive: false,
+  },
+  {
+    label: 'Location checking script',
+    prompt: 'Okay, my location is #{location}',
+    handler: function(params) {
+      if (params.location.toLowerCase() === 'nsw' || 
+          params.location.toLowerCase() === 'new south wales') {
+        return 'The most common plants in ' + params.location + 'are ...';
+      } else {
+        return ' I\'m sorry I couldnt find any plants in that area, maybe try the region?';
+        }
+    },
+  },
+  {
+    label: 'Location checking script',
+    prompt: 'My location is #{location}',
+    handler: function(params) {
+      if (params.location.includes(plants.locations)) {
+        return `The most common plant is ${plant.name}...`
+      } else {
+        return ' I\'m sorry I couldnt find any plants in that area, maybe try the region?'
+      }
+    },
+    isListening: true,
+    isCaseSensitive: false,
+  },
+  // I think i need a little explainer on how to get the above working. 
+  // BAsically what I'm trying to do is get the handler to check the object 'const plants = {...}' 
+  // then I want to use .includes() method to get the handler to check if the user prompt params includes
+  // any of the locations in the plants.locations string
+  // if it does, I want it to return the object or objects that includes the user location
+
+  // HALP i'm a bit lost still!
+
+  // I did a simialr thing with a search function... 
+
+  // in that case I stored the .value of the "search results" in const plantSearch
+  // then I .trim() and .toLowerCase() the result and stored it in const searchTrim
+  // then I checked if the users input is in any of the plants common names
+        // const isPlantName = (obj) => obj.commonName === searchTrim
+  // then created a huge if else tree to return the object that fit the user's input value
+        // ie.  if (plantName.includes(searchTrim)) {...}
+        // else if (plantLocation.includes(searchTrim)) {...}
+        //  ETC.
+        
+  //But i'm a little lost or too new to unpack how to do this here?
```

Comment:
I LOVE that you wrote down what you were struggling with and what you tried!
I also love where you are going with this! Beyond the humor, you provided the data that you'd like to reference so the user can genuinely find plants in their area! It is a beautiful objects of objects by the way!
I feel as though you are on to something and your syntax, even for just *attempting* these goals, is fantastic.